### PR TITLE
Clarify `val_check_interval` description

### LINKED ISSUE #7407: Confusion between steps and batches
Looking at `TrainLoop`,  it would appear that the global step takes into account `accumulate_grad_batches`:
https://github.com/PyTorchLightning/pytorch-lightning/blob/6104a6316afb8cdd9825d77844db456ffa766ca1/pytorch_lightning/trainer/training_loop.py#L884-L892

However, the documentation for the trainer makes it sound like steps and batches are equivalent:
https://github.com/PyTorchLightning/pytorch-lightning/blob/6104a6316afb8cdd9825d77844db456ffa766ca1/pytorch_lightning/trainer/trainer.py#L286-L287

1. Should the difference between steps and batches be clarified somewhere? Perhaps calling steps "optimizer steps" or "param updates" would help avoid any misunderstandings.
2. Additionally, when reporting metrics via e.g. tensorboard, would it not be better to have the _x_ axis show batches rather than steps? This would allow easy comparison between runs with different world sizes and different values of `accumulate_grad_batches`.

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes #7407

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [n/a] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @borda @rohitgr7

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

File: docs/source/common/trainer.rst [hunk 1]
```diff
@@ -1544,8 +1544,8 @@ val_check_interval
 How often within one training epoch to check the validation set.
 Can specify as float or int.
 
-- use (float) to check within a training epoch
-- use (int) to check every n steps (batches)
+- pass a ``float`` to check after a percentage of the training epoch
```
Comment (awaelchli) [inline docs/source/common/trainer.rst:1547]: I would not mention percentage because that's a value outside the range 0-1. Should we say "fraction" instead?
```suggestion
- pass a ``float`` in the range [0.0, 1.0] to check after a fraction of the training epoch
```
something like that?

File: pytorch_lightning/trainer/trainer.py [hunk 2]
```diff
@@ -389,8 +389,8 @@ def __init__(
             track_grad_norm: -1 no tracking. Otherwise tracks that p-norm. May be set to 'inf' infinity-norm. If using
                 Automatic Mixed Precision (AMP), the gradients will be unscaled before logging them.
 
-            val_check_interval: How often to check the validation set. Use float to check within a training epoch,
-                use int to check every n steps (batches).
+            val_check_interval: How often to check the validation set. Pass a ``float`` to check after a percentage of
```
Comment (awaelchli) [inline pytorch_lightning/trainer/trainer.py:392]: same comment as above

**Review by awaelchli — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]
